ideascube/monitoring/tests/test_migration_unique_serial.py

```diff
@@ -0,0 +1,67 @@
+from ideascube.tests.helpers import migration_test
+
+
+@migration_test(
+    migrate_from=[('monitoring', '0005_auto_20161027_0801')],
+    migrate_to=[('monitoring', '0006_unique_serial')]
+)
+def test_do_not_break_already_unique_serial(migration):
+    Specimen = migration.old_apps.get_model('monitoring', 'Specimen')
+    StockItem = migration.old_apps.get_model('monitoring', 'StockItem')
+
+    stockItem = StockItem()
+
+    for i in range(5):
+        specimen = Specimen(item_id=stockItem.id, barcode=str(i), serial=str(i))
+
+    migration.run_migration()
+
+    Specimen = migration.new_apps.get_model('monitoring', 'Specimen')
+    specimens = Specimen.objects.order_by('barcode')
+    for specimen in specimens:
+        assert specimen.barcode == specimen.serial
+
+
+@migration_test(
+    migrate_from=[('monitoring', '0005_auto_20161027_0801')],
+    migrate_to=[('monitoring', '0006_unique_serial')]
+)
+def test_create_unique_serial(migration):
+    Specimen = migration.old_apps.get_model('monitoring', 'Specimen')
+    StockItem = migration.old_apps.get_model('monitoring', 'StockItem')
+
+    stockItem = StockItem()
```

Comment:
In an earlier version you were saving the `stockItem` here. Why don't you do it any more?

More importantly, if the `stockItem` is not saved, then `stockItem.id` below iw `None`.